src/shared/creds-util.c

```diff
@@ -660,16 +660,15 @@ int encrypt_credential_and_warn(
 #if HAVE_TPM2
         bool try_tpm2;
         if (sd_id128_equal(with_key, _CRED_AUTO)) {
-                /* If automatic mode is selected and we are running in a container, let's not try TPM2. OTOH
-                 * if user picks TPM2 explicitly, let's always honour the request and try. */
+                /* If automatic mode is selected lets see if a TPM2 it is present. If we are running in a
+                 * container tpm2_support will detect this, and will return a different flag combination of
+                 * TPM2_SUPPORT_FULL, effectively skipping the use of TPM2 when inside one. */
 
-                r = detect_container();
-                if (r < 0)
-                        log_debug_errno(r, "Failed to determine whether we are running in a container, ignoring: %m");
-                else if (r > 0)
-                        log_debug("Running in container, not attempting to use TPM2.");
+                Tpm2Support support = tpm2_support();
 
-                try_tpm2 = r <= 0;
+                try_tpm2 = support == TPM2_SUPPORT_FULL;
```

Comment:
i'd really just do:

```c
try-tpm2 = tpm2_support() == TPM2_SUPPORT_FULL;
```
no need for the intermediary variable, this is clear enough and shorter

Reply:
Yes, it is more clear this way. Done.